storage/client/internal/parse_rfc3339.cc

```diff
@@ -0,0 +1,207 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/client/internal/parse_rfc3339.h"
+#include "google/cloud/internal/throw_delegate.h"
+#include <cctype>
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace {
+[[noreturn]] void ReportError(std::string const& timestamp, char const* msg) {
+  std::ostringstream os;
+  os << "Error parsing RFC 3339 timestamp: " << msg
+     << " Valid format is YYYY-MM-DD[Tt]HH:MM:SS[.s+](Z|+HH:MM), got="
+     << timestamp;
+  google::cloud::internal::RaiseInvalidArgument(os.str());
+}
+
+bool IsLeapYear(int year) {
+  return (year % 4 == 0 and (year % 100 != 0 or year % 400 == 0));
```

Comment:
Do the `and` and `or` keywords work without a special include on MSVC compilers?  Last I checked (it's been a few years since I was shackled to MSVC), you needed to include a special header or turn on a special compiler option for those symbols to be valid:
https://stackoverflow.com/questions/2376448/the-written-versions-of-the-logical-operators

Note: I personally prefer these over && and ||, and am all for including the header iso646 header (or whatever is the easiest option).

Reply:
I ran a quick test on https://godbolt.org/ and it seems they still don't work on MSVC 19 2017 without the iso646.h header.

Reply:
They do require the additional header, it is included in `version.h` which must be included everywhere (because that is where we define the namespace).

Reply:
Ah, I was looking for "iso646.h" -- it's there under "ciso646". Whoops :)